x/crosschain/keeper/msg_server_migrate_erc20_custody_funds.go

```diff
@@ -0,0 +1,104 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	authoritytypes "github.com/zeta-chain/zetacore/x/authority/types"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+// MigrateERC20CustodyFunds migrates the funds from the current TSS to the new TSS
+func (k msgServer) MigrateERC20CustodyFunds(
+	goCtx context.Context,
+	msg *types.MsgMigrateERC20CustodyFunds,
+) (*types.MsgMigrateERC20CustodyFundsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check if authorized
+	err := k.GetAuthorityKeeper().CheckAuthorization(ctx, msg)
+	if err != nil {
+		return nil, errorsmod.Wrap(authoritytypes.ErrUnauthorized, err.Error())
+	}
+
+	// get the current TSS nonce allow to set a unique index for the CCTX
+	chainNonce, found := k.GetObserverKeeper().GetChainNonces(ctx, msg.ChainId)
+	if !found {
+		return nil, errorsmod.Wrap(types.ErrInvalidChainID, "cannot find current chain nonce")
+	}
+	currentNonce := chainNonce.Nonce
+
+	// get the current TSS
+	tss, found := k.GetObserverKeeper().GetTSS(ctx)
+	if !found {
+		return nil, errorsmod.Wrap(types.ErrCannotFindTSSKeys, "cannot find current TSS")
+	}
+
+	// get necessary parameters to create the cctx
+	params, found := k.zetaObserverKeeper.GetChainParamsByChainID(ctx, msg.ChainId)
+	if !found {
+		return nil, errorsmod.Wrapf(types.ErrInvalidChainID, "chain params not found for chain id (%d)", msg.ChainId)
+	}
+	medianGasPrice, priorityFee, isFound := k.GetMedianGasValues(ctx, msg.ChainId)
+	if !isFound {
+		return nil, errorsmod.Wrapf(
+			types.ErrUnableToGetGasPrice,
+			"median gas price not found for chain id (%d)",
+			msg.ChainId,
+		)
+	}
+	// overpays gas price by 2x
+	const multiplier = 2
```

Comment:
Can we add this to the types of package under keys.go

Reply:
Added constant, however not under keys.go, this is the file for keys prefix, not related